src/main/java/lotto/Lotto.java

```diff
@@ -1,39 +1,67 @@
 package lotto;
 
 import java.util.List;
-import java.util.function.Supplier;
+import java.util.stream.Collectors;
 
 public class Lotto implements LottoTicket {
-	private final LottoNumbers lottoNumbers;
+	public static final String INVALID_NUMBER_COUNT = "잘못된 로또 숫자입니다.";
+	private final List<LottoNumber> lottoNumbers;
 	private final int price;
 	private final LottoRank rank;
 
-	public Lotto(Supplier<List<Integer>> lottoMaker, int lottoPrice) {
-		this.lottoNumbers = new LottoNumbers(lottoMaker.get());
+	public Lotto(List<Integer> lottoNumbers, int lottoPrice) {
+		if (!this.validate(lottoNumbers)) {
+			throw new IllegalArgumentException(INVALID_NUMBER_COUNT);
+		}
+		this.lottoNumbers = lottoNumbers
+			.stream()
+			.map(LottoNumber::new)
+			.collect(Collectors.toList());
 		this.price = lottoPrice;
 		this.rank = LottoRank.NOTHING;
 	}
 
-	public Lotto(LottoNumbers lottoNumbers, int price, LottoRank lottoRank) {
+	public Lotto(List<Integer> numbers) {
+		this(numbers, 0);
+	}
+
+	public Lotto(List<LottoNumber> lottoNumbers, int price, LottoRank lottoRank) {
 		this.lottoNumbers = lottoNumbers;
 		this.price = price;
 		this.rank = lottoRank;
 	}
 
-	public Lotto unmaskedLotto(Lotto other) {
-		return new Lotto(this.lottoNumbers, this.price, LottoRank.findLottoRank(this.lottoNumbers, other.lottoNumbers));
-	}
-
-	public LottoRank lottoRank() {
-		return this.rank;
+	private boolean validate(List<Integer> lottoNumbers) {
+		if (lottoNumbers.size() != LottoRule.LOTTO_NUMBER_COUNT) {
+			return false;
+		}
+		return true;
```

Comment:
if문 조건절 자체 return값이 boolean값이네요! :-) 
아래와 같이 표현 가능 하겠네요! 
```suggestion
		return lottoNumbers.size() == LottoRule.LOTTO_NUMBER_COUNT;
```